galaxy_ng/app/tasks/promotion.py

```diff
@@ -25,3 +26,35 @@ def remove_content_from_repository(collection_version_pk, repository_pk):
     qs = CollectionVersion.objects.filter(pk=collection_version_pk)
     with repository.new_version() as new_version:
         new_version.remove_content(qs)
+
+
+def dispatch_tasks_to_org_repos(collection_version_pk, action):
+    """Starts tasks to add/remove CollectionVersion to/from org repos.
+    Method to start tasks, not a task itself."""
+
+    # get all org distros
+    # TODO: replace with correct convention, regex in constants?
+    org_distros = AnsibleDistribution.objects.filter(name__startswith='org-')
+
+    # filter out distros which point to golden repo
+    golden = AnsibleRepository.objects.get(name='automation-hub')  # use constant or param
+    org_repos = set([d.repository for d in org_distros])
+    try:
+        org_repos.remove(golden)
+    except KeyError:
+        pass
+
```

Comment:
This may be a good place to use the pulpcore 'TaskGroup' stuff  
 https://docs.pulpproject.org/plugins/plugin-writer/concepts/index.html#tasks
https://github.com/pulp/pulpcore/blob/master/pulpcore/app/models/task.py#L423
https://docs.pulpproject.org/restapi.html#tag/task-groups

Reply:
ie, so their is a TaskGroup for all the tasks that are updating repos, and that TaskGroup has progress status showing how many of the tasks are done.